clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -924,7 +924,7 @@ private void handleTimeoutFailure(long now, Throwable cause) {
                 log.debug("{} timed out at {} after {} attempt(s)", this, now, tries,
                     new Exception(prettyPrintException(cause)));
             }
-            if (cause instanceof TimeoutException) {
+            if (cause instanceof TimeoutException || cause instanceof IllegalStateException) {
```

Comment:
I feel it's slightly confusing to handle `IllegalStateException` within `handleTimeoutException`. This might not be needed if we do what I suggested in the other comment.